### PR TITLE
User agreement and privacy policy

### PR DESCRIPTION
**Pull request by emanuele45**
Meant to cover the points in #3142 probably not always elegant, but still seems to work.

### CODE REVIEW

**Comment by Spuds** [comment]
From a look through, it seems good to me !

Is there any requirement to also accept cookies that we should add to this release?

**Comment by ahrasis** [comment]
Looks good to me too.

**Comment by emanuele45** [comment]
I would say no.
There is just one thing I forgot to add: during install and upgrade a "base" set of agreements should be created so that all the logic for approving with missing date could be removed.

File: themes/default/languages/english/Admin.english.php [hunk 1]
```diff
@@ -47,10 +47,14 @@
 $txt['admin_credits'] = 'Credits';
 $txt['admin_agreement'] = 'Show and require agreement letter when registering';
 $txt['admin_checkbox_agreement'] = 'Show a checkbox for the agreement in registration form instead of a full page';
+$txt['admin_checkbox_accept_agreement'] = 'Force all members to accept the agreement at the next visit to the forum';
 $txt['admin_agreement_default'] = 'Default';
 $txt['admin_agreement_select_language'] = 'Language to edit';
 $txt['admin_agreement_select_language_change'] = 'Change';
 
+$txt['admin_privacypol'] = 'Require to accept the privacy policy when registering';
```
Comment (Frenzie) [inline themes/default/languages/english/Admin.english.php]: I think `Require accepting` would sound better.

File: themes/default/languages/english/Admin.english.php [hunk 2]
```diff
@@ -124,6 +130,9 @@
 $txt['live'] = 'Latest Software Updates';
 $txt['remove_all'] = 'Clear Log';
 $txt['agreement_not_writable'] = 'Warning - agreement.txt is not writable, any changes you make will NOT be saved.';
+$txt['agreement_backup_not_writable'] = 'Warning - the backup directory in forum_root/packages/backup cannot be created.';
+$txt['privacypol_not_writable'] = 'Warning - privacypol.txt is not writable, any changes you make will NOT be saved.';
```
Comment (Frenzie) [inline themes/default/languages/english/Admin.english.php]: Comma splice

File: themes/default/languages/english/Errors.english.php [hunk 3]
```diff
@@ -428,4 +429,4 @@
 // Drag / Drop sort errors
 $txt['no_sortable_items'] = 'No sortable items were found';
 
-$txt['error_invalid_notification_id'] = 'An addons is trying to register a notification method with an existing ID. ID lower than 5 are protected and cannot be used by addons, if the ID is higher, than two addons may be sharing the same ID.';
\ No newline at end of file
+$txt['error_invalid_notification_id'] = 'An addons is trying to register a notification method with an existing ID. ID lower than 5 are protected and cannot be used by addons, if the ID is higher, than two addons may be sharing the same ID.';
```
Comment (Frenzie) [inline themes/default/languages/english/Errors.english.php]: * An addon
* IDs lower than 5
* Also a comma splice

`An addon is trying to register a notification method with an existing ID. IDs lower than 5 are protected and cannot be used by addons. If the ID is higher, then two addons may be sharing the same ID.`

File: themes/default/languages/english/Login.english.php [hunk 4]
```diff
@@ -28,6 +32,8 @@
 $txt['login_below'] = 'Please login below.';
 $txt['login_below_or_register'] = 'Please login below or <a href="%1$s">register an account</a> with %2$s';
 $txt['checkbox_agreement'] = 'I accept the registration agreement';
+$txt['checkbox_privacypol'] = 'I accept the privacy policy';
+$txt['confirm_request_accept_agreement'] = 'Are you sure you want to force all the users to accept the agreement?';
```
Comment (Frenzie) [inline themes/default/languages/english/Login.english.php]: `force all users` (no determiner)

File: themes/default/languages/english/Login.english.php [hunk 5]
```diff
@@ -147,4 +153,4 @@
 $txt['contact_your_message'] = 'Your message';
 $txt['errors_contact_form'] = 'The following errors occurred while processing your contact request';
 $txt['contact_subject'] = 'A guest has sent you a message';
-$txt['contact_thankyou'] = 'Thank you for your message, someone will contact you as soon as possible.';
\ No newline at end of file
+$txt['contact_thankyou'] = 'Thank you for your message, someone will contact you as soon as possible.';
```
Comment (Frenzie) [inline themes/default/languages/english/Login.english.php]: Comma splice

File: themes/default/languages/english/index.english.php [hunk 6]
```diff
@@ -550,6 +550,7 @@
 $txt['not_removed_extra'] = '%1$s is a backup of %2$s that was not generated by ElkArte. It can be accessed directly and used to gain unauthorised access to your forum. You should delete it immediately.';
 $txt['generic_warning'] = 'Warning';
 $txt['agreement_missing'] = 'You are requiring new users to accept a registration agreement, however the file (agreement.txt) doesn\'t exist.';
+$txt['agreement_accepted'] = 'You have successfully accepted the agreement.';
```
Comment (Frenzie) [inline themes/default/languages/english/index.english.php]: That "successfully" seems a bit redundant?

File: themes/default/languages/english/Admin.english.php [hunk 7]
```diff
@@ -129,9 +129,9 @@
 $txt['attachment_delete_admin'] = '[attachment deleted by admin]';
 $txt['live'] = 'Latest Software Updates';
 $txt['remove_all'] = 'Clear Log';
-$txt['agreement_not_writable'] = 'Warning - agreement.txt is not writable, any changes you make will NOT be saved.';
+$txt['agreement_not_writable'] = 'Warning - agreement.txt is not writable any changes you make will NOT be saved.';
```
Comment (Frenzie) [inline themes/default/languages/english/Admin.english.php]: Oh no, the comma should be replaced by a period instead. ;-)

Comment (Frenzie) [reply themes/default/languages/english/Admin.english.php]: Or you could insert a "so" but I think two smaller sentences is probably better stylistically.

**Comment by emanuele45** [comment]
I would say that that's all for this PR.

File: themes/default/languages/english/Login.english.php [hunk 8]
```diff
@@ -28,6 +32,9 @@
 $txt['login_below'] = 'Please login below.';
 $txt['login_below_or_register'] = 'Please login below or <a href="%1$s">register an account</a> with %2$s';
 $txt['checkbox_agreement'] = 'I accept the registration agreement';
+$txt['checkbox_privacypol'] = 'I accept the privacy policy';
+$txt['confirm_request_accept_agreement'] = 'Are you sure you want to force all members to accept the agreement?';
+$txt['confirm_request_accept_privacy_policy'] = 'Are you sure you want to force all members to accept the prvacy policy?';
```
Comment (Frenzie) [inline themes/default/languages/english/Login.english.php]: typo

File: themes/default/languages/english/Login.english.php (hunk 5 above)
Comment (Frenzie) [reply themes/default/languages/english/Login.english.php]: Thank you for your message. Someone will contact you as soon as possible.

**Comment by Frenzie** [comment]
@emanuele45 Great work!

**Comment by ahrasis** [comment]
Can't wait for 1.1.4. ;)

**Comment by emanuele45** [comment]
@Spuds any idea why 5.4 is failing? :confused:

**Comment by Frenzie** [comment]
@emanuele45 Unless you mean the one here that's failing because you deleted the branch, looks like a clear Travis/Amazon AWS issue. ;-)

```
5.4 is not pre-installed; installing
Downloading archive: https://s3.amazonaws.com/travis-php-archives/binaries/ubuntu/14.04/x86_64/php-5.4.tar.bz2
127.33s$ curl -s -o archive.tar.bz2 $archive_url && tar xjf archive.tar.bz2 --directory /
curl: (7) Failed to connect to s3.amazonaws.com port 443: Connection timed out
```

**Pull request merged**